unified_planning/solvers/grounder.py

```diff
@@ -44,35 +52,44 @@ def name(self):
-        supported_kind.set_fluents_type('NUMERIC_FLUENTS')
-        supported_kind.set_fluents_type('OBJECT_FLUENTS')
-        supported_kind.set_conditions_kind('NEGATIVE_CONDITIONS')
-        supported_kind.set_conditions_kind('DISJUNCTIVE_CONDITIONS')
-        supported_kind.set_conditions_kind('EQUALITY')
-        supported_kind.set_conditions_kind('EXISTENTIAL_CONDITIONS')
-        supported_kind.set_conditions_kind('UNIVERSAL_CONDITIONS')
-        supported_kind.set_effects_kind('CONDITIONAL_EFFECTS')
-        supported_kind.set_effects_kind('INCREASE_EFFECTS')
-        supported_kind.set_effects_kind('DECREASE_EFFECTS')
-        supported_kind.set_time('CONTINUOUS_TIME')
-        supported_kind.set_time('DISCRETE_TIME')
-        supported_kind.set_time('INTERMEDIATE_CONDITIONS_AND_EFFECTS')
-        supported_kind.set_time('TIMED_EFFECT')
-        supported_kind.set_time('TIMED_GOALS')
-        supported_kind.set_time('DURATION_INEQUALITIES')
-        supported_kind.set_simulated_entities('SIMULATED_EFFECTS')
-        return problem_kind <= supported_kind
+        supported_kind.set_typing('FLAT_TYPING') # type:ignore
+        supported_kind.set_typing('HIERARCHICAL_TYPING') # type:ignore
+        supported_kind.set_numbers('CONTINUOUS_NUMBERS') # type:ignore
+        supported_kind.set_numbers('DISCRETE_NUMBERS') # type:ignore
+        supported_kind.set_fluents_type('NUMERIC_FLUENTS') # type:ignore
+        supported_kind.set_fluents_type('OBJECT_FLUENTS') # type:ignore
+        supported_kind.set_conditions_kind('NEGATIVE_CONDITIONS') # type:ignore
+        supported_kind.set_conditions_kind('DISJUNCTIVE_CONDITIONS') # type:ignore
+        supported_kind.set_conditions_kind('EQUALITY') # type:ignore
+        supported_kind.set_conditions_kind('EXISTENTIAL_CONDITIONS') # type:ignore
+        supported_kind.set_conditions_kind('UNIVERSAL_CONDITIONS') # type:ignore
+        supported_kind.set_effects_kind('CONDITIONAL_EFFECTS') # type:ignore
+        supported_kind.set_effects_kind('INCREASE_EFFECTS') # type:ignore
+        supported_kind.set_effects_kind('DECREASE_EFFECTS') # type:ignore
+        supported_kind.set_time('CONTINUOUS_TIME') # type:ignore
+        supported_kind.set_time('DISCRETE_TIME') # type:ignore
+        supported_kind.set_time('INTERMEDIATE_CONDITIONS_AND_EFFECTS') # type:ignore
+        supported_kind.set_time('TIMED_EFFECT') # type:ignore
+        supported_kind.set_time('TIMED_GOALS') # type:ignore
+        supported_kind.set_time('DURATION_INEQUALITIES') # type:ignore
+        supported_kind.set_simulated_entities('SIMULATED_EFFECTS') # type:ignore
+        return supported_kind
+
+    @staticmethod
+    def supports(problem_kind):
+        return problem_kind <= Grounder.supported_kind()
 
     @staticmethod
     def is_grounder():
         return True
 
+    @staticmethod
+    def credits(stream: Optional[IO[str]] = sys.stdout, full_credits: bool = False):
+        if stream is not None:
```

Comment:
Why not moving this `if` into the caller and have this function take a non-optional `IO[str]`?